### PR TITLE
No schema files #159

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java [hunk 1]
```diff
@@ -6,32 +6,3565 @@
...
+
+        @Override
+        public <T> T getObject(String columnLabel, Class<T> type) throws SQLException {
+
+            return null;
+        }
+
+    }
+
+    public class PrintCallableStatement implements java.sql.CallableStatement {
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Is it really needed?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java [hunk 2]
```diff
@@ -6,32 +6,3565 @@
...
+
+        @Override
+        public boolean isCloseOnCompletion() throws SQLException {
+
+            return false;
+        }
+
+    }
+
+    public class PrintResultSet implements java.sql.ResultSet {
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Is it really needed?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java [hunk 3]
```diff
@@ -6,32 +6,3565 @@
...
+
+        @Override
+        public int getNetworkTimeout() throws SQLException {
+
+            return 0;
+        }
+
     }
+
+    public class PrintPreparedStatement implements java.sql.PreparedStatement {
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Is it really needed?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java [hunk 4]
```diff
@@ -6,32 +6,3565 @@
...
+        } catch (FileNotFoundException e) {
+            System.out.println("The File is not able to be created or found - [" + outputFile + "]");
+        }
+    }
+
+    /*
+     * The following classes are 'dummy implementations' that enable printing of the
+     * SQL.
+     */
+    public class PrintConnection implements java.sql.Connection {
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Is it really needed?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java [hunk 5]
```diff
@@ -6,32 +6,3565 @@
...
+            printer.processApplyGrants();
+            printer.printGrants();
+        } catch (FileNotFoundException e) {
+            System.out.println("The File is not able to be created or found - [" + outputFile + "]");
+        }
+    }
+
+    /*
+     * The following classes are 'dummy implementations' that enable printing of the
+     * SQL.
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: why do we need the dummy stubs?  no way to just use the existing Print classes?
if its really needed, maybe we introduce test dependency for creating the stubs from the Interface?

**Comment by lmsurpre** [comment]
do we really want to commit to keeping the DDL up-to-date in our docs?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 2 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Yes, so we get beyond the rs.next in the SELECT that checks for the version history table

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 1 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: I wanted to be able to cover future changes and capture them as part of the print. 
There are some db2 commands which can be executed as callables.

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 3 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Yes - to get the sql

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 4 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Yes - to get the sql (via the related PreparedStatement)

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 5 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: existing print classes miss many commands and do not fully include all the commands

**Comment by prb112** [comment]
> do we really want to commit to keeping the DDL up-to-date in our docs?

Perfectly valid question... maybe teach questions/decision?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java (hunk 5 above)
Comment (lmsurpre) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: can we fix those?

Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java]: Not really.  It'll be a pain to re-engineer to support it; I looked at it.  This is the reliable way - regardless of IDatabaseTarget and IDatabaseTranslator.

File: fhir-persistence-schema/README.md [hunk 6]
```diff
@@ -83,4 +83,31 @@ Edit `wlp/usr/servers/fhir-server/config/TNT1/fhir-server-config.json` and add t
 --update-proc
 ```
 
+# Printing the Schema
+
+To run this code, build the jar (fhir-database-utils and fhir-persistence-schema) <br>
```
Comment (lmsurpre) [inline fhir-persistence-schema/README.md]: remove the <br>
```suggestion
To run this code, build the jar (fhir-database-utils and fhir-persistence-schema)
```

File: fhir-persistence-schema/README.md [hunk 7]
```diff
@@ -83,4 +83,31 @@ Edit `wlp/usr/servers/fhir-server/config/TNT1/fhir-server-config.json` and add t
...
+
+``` shell 
+java -cp ./fhir-database-utils.jar:fhir-persistence-schema.jar com.ibm.fhir.schema.app.SimpleSchemaPrinter [--to-file]<br>
```
Comment (lmsurpre) [inline fhir-persistence-schema/README.md]: remove the `<br>`
```suggestion
java -cp ./fhir-database-utils.jar:fhir-persistence-schema.jar com.ibm.fhir.schema.app.SimpleSchemaPrinter [--to-file]
```

File: fhir-persistence-schema/README.md [hunk 8]
```diff
@@ -83,4 +83,31 @@ Edit `wlp/usr/servers/fhir-server/config/TNT1/fhir-server-config.json` and add t
...
+For db2 import to
+- schema.sql `db2 -tvf schema.sql`
+- grants.sql `db2 -tvf grants.sql`
+- stored-procedures.sql `db2 -td@ -vf stored-procedures.sql`
+
+The schema is then available for use. 
+
+# Creating the database 
+
+Create the database and database user, please use the following command.
```
Comment (lmsurpre) [inline fhir-persistence-schema/README.md]:
```suggestion
To create the database and database user, please use the following command.
```

**Comment by prb112** [comment]
got it

**Pull request merged**